app/src/main/java/org/eyeseetea/malariacare/domain/usecase/pull/DownloadMediaUseCase.java

```diff
@@ -0,0 +1,259 @@
+                                        REQUEST_AUTHORIZATION);
+                                return;
+                            }
+
+                            //Real connection google error
+                            if (mLastError instanceof GooglePlayServicesAvailabilityIOException) {
+                                showGooglePlayServicesAvailabilityErrorDialog(
+                                        ((GooglePlayServicesAvailabilityIOException) mLastError)
+                                                .getConnectionStatusCode());
+                                return;
+                            }
+
+                            //Other error
+                            Log.e(TAG, "onCancelled: " + mLastError == null ? "" : mLastError.getMessage());
+                        }
+
+                        @Override
+                        public void onSuccess(HashMap<String, String> syncedFiles) {
+                            int correctSyncedFiles=0;
+
+                            //Try to reuse a local copy if another media references same url
+                            if(syncedFiles.size()>0){
+                                for(Media media:medias){
+                                    String absolutePath = syncedFiles.get(media.getResourceUrl());
+                                    List<Media> allMediaWithSameResource = mMediaRepository.getAllMediaByResourceUid(media.getResourceUrl());
+                                    for(Media localMedia:allMediaWithSameResource) {
+                                        Log.d(TAG, localMedia.toString()+ "\tsaved in " + absolutePath);
+                                        localMedia.setResourcePath(absolutePath);
+                                        mMediaRepository.updateModel(localMedia);
+                                    }
+                                }
+
+                            }
+                            if(correctSyncedFiles>0) {
+                                showMessage(String.format("%d files synced", syncedFiles.size()));
+                            }
+                        }
+
+                        @Override
+                        public void removeIllegalFile(String uid) {
+                            Log.e(TAG, "downloadFile error (file extension not supported)" + uid);
+                            mMediaRepository.deleteModel(mMediaRepository.findByUid(uid));
+                        }
+                    }).execute();
+        }
+    }
+
+    private void initServiceAccountCredential() {
+        try {
+            InputStream privateJsonStream = mContext.getResources().openRawResource(
+                    R.raw.driveserviceprivatekey);
+            serviceCredential = GoogleCredential.fromStream(privateJsonStream).createScoped(
+                    Arrays.asList(SCOPES));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private boolean isGooglePlayAppAvailable() {
```

Comment:
this is a Google Drive service requirement and not a domain requirement then this validation should be realized in new FileDownloader adapter to create.